app/views/layouts/_listicon.html.haml

```diff
@@ -1,139 +1,5 @@
-    %li
-      %span.fa.fa-clock-o
-- elsif db == "MiqGroup"
-  = image_tag("#{img_path}new/miq_group.png", img_attr_20)
-- elsif db == "MiqReportResult"
-  - case row['status'].downcase
-  - when "error"
-    - glyph = "fa fa-warning"
-  - when "finished"
-    - glyph = "pficon pficon-ok"
-  - when "running"
-    - glyph = "fa fa-play"
-  - when "queued"
-    - glyph = "fa fa-pause"
-  - else
-    - glyph = "product product-arrow-right"
-  %ul.icons.list-unstyled
-    %li
-      %span{:class => glyph}
-- elsif db == "MiqUserRole"
-  %ul.icons.list-unstyled
-    %li
-      %span{:class => "product product-role"}
-- elsif db == "MiqWidget"
-  - case row['content_type'].downcase
-  - when "chart"
-    - glyph = "product product-chart"
-  - when "menu"
-    - glyph = "fa fa-share-square-o"
-  - when "report"
-    - glyph = "product product-report"
-  - when "rss"
-    - glyph = "fa fa-rss"
-  %ul.icons.list-unstyled
-    %li
-      %span{:class => glyph}
-      - status = row.status
-      - case status.downcase
-      - when "complete"
-        - glyph = "pficon pficon-ok"
-      - when "queued"
-        - glyph = "fa fa-pause"
-      - when "running"
-        - glyph = "fa fa-play"
-      - when "error"
-        - glyph = "fa fa-warning"
-      - if status.downcase != "none"
-        %span{:class => glyph}
-- elsif db == "MiqWorker"
-  - worker = @targets_hash[from_cid(@id)] if @targets_hash
-  = image_tag("#{img_path}new/processmanager-#{worker.normalized_type}.png", img_attr_20)
-- elsif db == "ScanItemSet"
-  = image_tag("#{img_path}new/scan_item_set.png", img_attr_20)
-- elsif db == "ChargebackRate"
-  = image_tag("#{img_path}new/chargeback_rates.png", img_attr_20)
-- elsif db == "ServerBuild"
-  = link_to(image_tag("/images/toolbars/delete.png",
-                      :class => "rollover small",
-                      :alt   => _("Click to delete this build")),
-            {:action => "delete_build", :id => @id},
-            "data-miq_sparkle_on"  => true,
-            "data-miq_sparkle_off" => true,
-            :confirm               => _("Are you sure you want to delete build %s?") % row.build,
-            :remote                => true,
-            :title                 => _("Click to delete this build"))
-- elsif db == "MiqProvision"
-  = image_tag("#{img_path}new/miq_request.png", img_attr_20)
-- elsif %w(CloudTenant CimBaseStorageExtent LdapRegion OntapStorageSystem OntapLogicalDisk OntapStorageVolume OntapFileShare OrchestrationStack OrchestrationTemplate SniaLocalFileSystem).include?(db)
-  = image_tag("#{img_path}new/#{h(db.underscore)}.png", img_attr_20)
-- elsif db == "ServiceTemplate"
-  - st = @targets_hash[from_cid(@id)] if @targets_hash
-  - img = st.respond_to?(:picture) && st.picture ? "../../../pictures/#{st.picture.basename}" : "/images/icons/new/service_template.png"
-  = image_tag(img, img_attr_20)
+- if %w(MiqReportResult MiqSchedule MiqUserRole MiqWidget).include?(db)
+  = listicon_glyphicon_tag(db, row)
```

Comment:
@chessbyte db is the type of list view being viewed, in most cases we use db to determine which image to show on the list view, in some specific cases we render different images. row is the actual record, we need to pass that in for the list views where we show images depending upon status or maybe other fields of each record.